hazelcast/src/main/java/com/hazelcast/logging/Logger.java

```diff
@@ -29,97 +35,91 @@ private Logger() {
     }
 
     /**
-     * @param clazz class to take the logger name from
-     * @return the logger
-     * Use LoggingService instead if possible. Do not log in static context if possible.
+     * Obtains a {@link ILogger logger} for the given {@code clazz}.
+     *
+     * @param clazz the class to obtain the logger for.
+     * @return the obtained logger.
      */
     public static ILogger getLogger(Class clazz) {
         return getLogger(clazz.getName());
     }
 
     /**
-     * @param name name of the logger
-     * @return the logger
-     * Use LoggingService instead if possible. Do not log in static context if possible.
+     * Obtains a {@link ILogger logger} of the given {@code name}.
+     *
+     * @param name the name of the logger to obtain.
+     * @return the obtained logger.
      */
     public static ILogger getLogger(String name) {
-        LoggerFactory loggerFactoryToUse = loggerFactory;
-        // fast-track when factory is initialized
-        if (loggerFactoryToUse != null) {
-            return loggerFactoryToUse.getLogger(name);
-        }
-
-        synchronized (FACTORY_LOCK) {
-            if (loggerFactory == null) {
-                // init static logger with user-specified custom logger class
-                String loggerClass = System.getProperty("hazelcast.logging.class");
-                if (!StringUtil.isNullOrEmpty(loggerClass)) {
-                    // ok, there is a custom logging factory class -> let's use it now and store it for next time
-                    loggerFactoryToUse = loadLoggerFactory(loggerClass);
-                    loggerFactory = loggerFactoryToUse;
-                } else {
-                    String loggerType = System.getProperty("hazelcast.logging.type");
-                    loggerFactoryToUse = newLoggerFactory(loggerType);
-                    // store the factory only when the type was set explicitly. we do not want to store default type.
-                    if (!StringUtil.isNullOrEmpty(loggerType)) {
-                        loggerFactory = loggerFactoryToUse;
-                    }
-                }
-            }
-        }
-        // loggerFactory was initialized by other thread before we acquired the lock -> loggerFactoryToUse was left to null
-        if (loggerFactoryToUse == null) {
-            loggerFactoryToUse = loggerFactory;
-        }
-        return loggerFactoryToUse.getLogger(name);
+        return getOrCreateLoggerFactory(System.getProperty("hazelcast.logging.type")).getLogger(name);
```

Comment:
Can you spread this over multiple lines. I'm not crazy about such complex expressions because it makes code harder to understand.